HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,244 @@
+        // Save Quaternion --> Transformation Matrix to .YAML file
+        auto fileStorageOut = cv::FileStorage{};
+        if(!fileStorageOut.open("robotTransformOut.yaml", cv::FileStorage::Mode::WRITE))
+        {
+            throw std::runtime_error("Could not open robotTransformOut.yaml for writing");
+        }
+        fileStorageOut.write("TransformationMatrixFromQuaternion",
+                             eigenToCv(transformationMatrixFromQuaternion.matrix()));
+        fileStorageOut.release();
+    }
+
+    catch(const std::exception &e)
+    {
+        std::cerr << "Error: " << e.what() << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+cv::Mat eigenToCv(const Eigen::MatrixXd &eigenMat)
+{
+    cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+    cv::eigen2cv(eigenMat, cvMat);
+
+    return cvMat;
+}
+
+Eigen::MatrixXd cvToEigen(const cv::Mat &cvMat)
+{
+    Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+    cv::cv2eigen(cvMat, eigenMat);
+
+    return eigenMat;
+}
+
+Representations transformFromZividToRobot(const Eigen::Affine3d &transformationMatrix)
+{
+    Representations robotRepresentations;
+    std::cout << "\nConverting Rotation Matrix to Axis-Angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle(transformationMatrix.linear());
+    std::cout << "Axis:\n" << axisAngle.axis() << std::endl;
+    std::cout << "Angle:\n" << axisAngle.angle() << std::endl;
+
+    // Axis-angle to Rotation Vector
+    std::cout << "\nConverting Axis-Angle to Rotation Vector:" << std::endl;
+    if(!axisAngle.axis().isUnitary())
+    {
+        std::cout << "Note! Axis in Axis-Angle pair is not unitary" << std::endl;
+    }
+    robotRepresentations.rotationVector = axisAngle.angle() * axisAngle.axis();
+    std::cout << robotRepresentations.rotationVector << std::endl;
+
+    // Rotation Matrix to Quaternion
+    std::cout << "\nConverting Rotation Matrix to Quaternion:" << std::endl;
+    const Eigen::Quaterniond quaternion(transformationMatrix.linear());
+    robotRepresentations.quaternion = quaternion;
+    std::cout << robotRepresentations.quaternion.coeffs() << std::endl;
+
+    // Rotation Matrix to Roll-Pitch-Yaw
+    for(int i = 0; i < nofRotationConventions; i++)
+    {
+        const RotationConvention convention{ static_cast<RotationConvention>(i) };
+        std::cout << "\nConverting Rotation Matrix to Roll-Pitch-Yaw angles (" << toString(convention)
+                  << "):" << std::endl;
+        robotRepresentations.rotations[i] = { static_cast<RotationConvention>(i),
+                                              rotmat2rpy(transformationMatrix.linear(), convention) };
+        std::cout << robotRepresentations.rotations[i].rollPitchYaw << std::endl;
+    }
+
+    return robotRepresentations;
+}
+
+Eigen::Affine3d transformFromRobotToZivid(const Eigen::Affine3d &transformationMatrix,
```

Comment:
Is it possible to replace transformationMatrix with translationVector ( transformationMatrix.translation() ) as the input parameter? It is confusing that you need rotationsRepresentation and one more rotation representation(inside transformationMatrix)

Reply:
Again, its just because I believe it looks more elegant to pass around a reference to the parent object, not its members.
I can change it.

Reply:
@knatten what is your opinion? Maybe what I am suggesting is not considered good programming? My reasoning is not to pass to a function something that is not completely used.